src/include/storage/sinval.h

```diff
@@ -80,6 +80,14 @@ typedef struct
 	int8		id;				/* type field --- must be first */
 	Oid			dbId;			/* database ID, or 0 if a shared relation */
 	Oid			relId;			/* relation ID, or 0 if whole relcache */
+	/* 
+	 * We need a way to mark whether this inval message is for a local ENR entry,
+	 * so that we can skip inserting it into the SI message queue.
+	 * Note that checking whether the relId is an ENR entry at the time of insertion
+	 * is insufficient, as during drops the ENR entry is deleted before we perform
+	 * SI queue insertion.
+	 */
+	bool		local_only;
```

Comment:
how are we making sure that this variable is false in all code paths. We will start with uninitialised value

Reply:
The only code path which sets this value is inside of `AddRelcacheInvalidationMessage()` which is guaranteed to set the value to true or false. I have done a full search of the code base, and no other code assigns any values to `msg.rc`.